CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Ephemeral Namespace Operator Local Development
+
+### Purpose
+The purpose of this doc is to deploy the Ephemeral Namespace Operator (ENO) in Minikube for local development.  
+
+### Requirements
+- Clone the [Clowder](https://github.com/RedHatInsights/clowder) repository
+- Install [Minikube](https://minikube.sigs.k8s.io/docs/start/)
+- Login credentials for [Quay](https://quay.io)
+- Install [Bonfire](https://pypi.org/project/crc-bonfire/)
+
+### Setup Minikube
+```
+minikube start --cpus 4 --disk-size 36GB --memory 8192MB --addons registry --addons ingress --addons metrics-server
+```
+
+### Retrieve Quay Pull Secret
+The following steps with walk you through getting your quay pull secret:
+- Login to [Quay](https://quay.io)
+- Click on your username at the top right of the homepage and click `Account Settings`
+- At the `CLI Password` section, click `Generate Encrypted Password`
+- Click `Kubernetes Secret`
+- Click `Download <your username>-secret.yml`
+
+### Setup Ephemeral Base Namespace
+The `ephemeral-base` namespace is used for copying secrets to the ephemeral namespaces that are created using the ENO.
+
+Run the following command to create the namespace:
+```
+oc create ns ephemeral-base --dry-run=client -o yaml | oc apply -f -
+```
+
+Apply the pull secret yaml file you retrieved from Quay:
+```
+oc apply -f ~/path/to/<your username>-secret.yml -n ephemeral-base
+```
+
+### Install Clowder
+Change to the Clowder directory and run:
+```
+./build/kube_setup.sh && make deploy-minikube-quick
+```
+
+### Run the Ephemeral Namespace Operator
+Change to the ENO directory and apply the following CRD:
+```
+oc apply -f config/crd/static/cloud.redhat.com_frontendenvironments.yaml
+```
+
+Start the ENO:
+```
+make run
+```
+
+You will need to use multiple terminal windows.
+
+### Apply Ephemeral Namespace Operator Pool Spec
+In the root directory of the Ephemeral Namespace Operator there is a pool yaml file for testing called `test-pool-spec.yaml`.  
+Ensure to edit line 56 of the file to update the name of the pull secret to the one retrieved from Quay. You will want to apply the  
+file to your Minikube cluster with the following command:  
+```
+oc apply -f ~/path/to/test-pool-spec.yaml
+```
+
+You can have multiple files with different configurations which will create various pool objects. This will spin up the number  
```

Comment:
Perhaps it's also a good idea here to describe what this pool is and how the ClowdEnvironment is related. Talking a little baout how the modes are set to `none` or close to, so that we don't use up so many resources wouldn't be a bad idea. The instruction to change the name of the pool, is that necessary if you only want to use a single pool? The instruction reads like you should always change the name, which I don't think is accurate.

If you are going to suggest changing the size I'd give them a little information as to why. Mention that the size attribute is how many of that particular namespace will be made available at a minimum and that in order to test multiple pools together, you probably want to reduce it.